nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java

```diff
@@ -169,7 +169,7 @@ public ConfigValue getConfigValue(String propertyName) {
     @Override
     public <T> Optional<List<T>> getOptionalValues(String propertyName, Class<T> propertyType) {
         Optional<List<T>> valuesList =  Config.super.getOptionalValues(propertyName, propertyType);
-        if (valuesList.get().isEmpty()) {
+        if (valuesList.isPresent() && valuesList.get().isEmpty()) {
```

Comment:
How about using Optional.filter?

Reply:
Thanks for kindly reminding that method!
I've updated the implementation.